### PR TITLE
Update view

### PR DESCRIPTION
**Pull request by spiresheep**

### CODE REVIEW

File: source/main.tsx [hunk 1]
```diff
@@ -33,6 +33,23 @@ const someStyle = {
   }
 };
 
+function changeValues() {
+  const rowsToChange = Math.floor(Math.random() * model.rowCount);
+  console.log(rowsToChange);
```
Comment (kranar) [inline source/main.tsx]: Remove logging.

File: source/table_view.tsx [hunk 2]
```diff
@@ -135,6 +137,9 @@ export class TableView extends React.Component<Properties> implements
     this.headerRowRef.style.cursor = 'auto';
   } 
 
+
+
```
Comment (kranar) [inline source/table_view.tsx]: Don't add new lines.

File: source/table_view.tsx [hunk 3]
```diff
@@ -48,6 +49,7 @@ export class TableView extends React.Component<Properties> implements
       this.columnResizer.onMouseMove.bind(this.columnResizer));
   }
 
+
```
Comment (kranar) [inline source/table_view.tsx]: Don't add new lines.

File: source/table_view.tsx [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { css, StyleSheet } from 'aphrodite/no-important';
+import * as Kola from 'kola-signals';
```
Comment (kranar) [inline source/table_view.tsx]: Where is this used?

File: source/table_view.tsx [hunk 5]
```diff
@@ -36,6 +36,7 @@ export class TableView extends React.Component<Properties> implements
     for(let i = 0; i < this.props.labels.length; ++i) {
       this.headerRefs[i] = null;
     }
+    this.props.model.connect(() => {this.forceUpdate();});
```
Comment (kranar) [inline source/table_view.tsx]: Use bind.

**Review by kranar — CHANGES_REQUESTED** [review]

File: source/table_view.tsx [hunk 6]
```diff
@@ -36,7 +35,7 @@ export class TableView extends React.Component<Properties> implements
     for(let i = 0; i < this.props.labels.length; ++i) {
       this.headerRefs[i] = null;
     }
-    this.props.model.connect(() => {this.forceUpdate();});
+    this.props.model.connect(this.handleUpdate.bind(this));
```
Comment (kranar) [inline source/table_view.tsx]: Don't add another method.

File: source/table_view.tsx [hunk 7]
```diff
@@ -35,7 +35,7 @@ export class TableView extends React.Component<Properties> implements
     for(let i = 0; i < this.props.labels.length; ++i) {
       this.headerRefs[i] = null;
     }
-    this.props.model.connect(this.handleUpdate.bind(this));
+    this.props.model.connect(this.setState.bind(this));
```
Comment (kranar) [inline source/table_view.tsx]: You went from ```() => {this.forceUpdate();}``` to adding a new method that wraps ```this.forceUpdate();``` to now using ```this.setState,bind(this);```. All of these are incorrect in one way or another. The correct solution is simply:

```this.props.model.connect(this.forceUpdate.bind(this));```

Comment (spiresheep) [reply source/table_view.tsx]: That doesn't work. I wanted to do that but the code keeps crashing.

Comment (kranar) [reply source/table_view.tsx]: Okay good point then, what happens specifically?

Comment (spiresheep) [reply source/table_view.tsx]: Everything is great until changeValues is called.

There is a warning: Warning: forceUpdate(...): Expected the last optional `callback` argument to be a function. Instead received: [object Object].

The error messages is: Error: Invalid argument passed as callback. Expected a function. Instead received: [object Object]

Comment (kranar) [reply source/table_view.tsx]: Okay that's a good catch, ```forceUpdate``` takes a callback parameter, so you will need to bind it to a null callback as follows:

```this.forceUpdate.bind(this, null);```

Comment (spiresheep) [reply source/table_view.tsx]: That explains it. I'll fix it now.

**Review by kranar — APPROVED** [review]

**Pull request merged**